src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp

```diff
@@ -675,7 +690,15 @@ namespace OpenMS
 
     std::vector<AccurateMassSearchResult> results_part;
 
-    queryByMZ(feature.getMZ(), feature.getCharge(), ion_mode, results_part);
+    String use_feature_adducts = param_.getValue("use_feature_adducts").toString();
+    if ((use_feature_adducts == "true") && feature.metaValueExists("dc_charge_adducts"))
```

Comment:
not sure if this is needed, but you could convert to bool.
bool use_feature_adducts = (param_.getValue("use_feature_adducts").toString() == "true") ? true : false;
if ((use_feature_adducts) && ..